### ISSUE TITLE
Error when adding video to "My List"

### ISSUE THREAD

**Issue by knappster1** [asciidisco/plugin.video.netflix#383]
*I'm submitting a ...*
  - [ x] bug report
  - [ ] feature request
  - [ ] support request

### Prerequisites

* [x ] Are you running the latest version?

### Description

When selecting a video and adding to My List, an error message pops up.

### Steps to Reproduce

1. Select Video
2. Add to My List

**Expected behavior:** [What you expected to happen]
Video would be added to My List in Netflix
**Actual behavior:** [What actually happened]
Error message and video is not added to My List in Netflix
### Installation

* [x ] I installed the plugin via zip from the Releases page
* [x ] I´m using the Netflix Repo
* [ ] I´m using a different source (Please tell which)

### Operating System

* [ x] Linux (x86/x64)
* [ ] OSX (x86/x64)
* [ ] Windows (x86/x64)
* [ ] Linux (ARM)
* [ ] Android

#### Additional informatin on the environment

Arch based Linux with Kodi 18 alpha 1

## Debug log

https://pastebin.com/0dD8AdJN

**Comment by Dodobibi** [comment]
It seems than Netflix refuse your request : Error 401, not authorized, then response is empty...
Are you try to unlogged and reauth from the addon ?

**Comment by mattreecebentley** [comment]
Also some strange behaviour when removing items from My List, variable as to effect.
Best case was, item moved to bottom of my list, then disappeared next time list was viewd.
Worst case was a crash.

**Comment by knappster1** [comment]
Sorry, I've been away a few days @Dodobibi .  I will try to logout and login again to see if that corrects it.  I should also note that it was 0.13.6 and I have not yet tried 0.13.7

**Comment by knappster1** [comment]
I've just tried logging out and back in and confirmed the same thing happens.  I've just updated to 0.13.7 as well with the same results.  It errors for both adding and removing videos from my list.  Are there any other tasks I could try that might lead to some useful troubleshooting information?

**Comment by knappster1** [comment]
@Dodobibi I thought I would check on this since there hasn't been activity for a while.  Are others having issues with "My List" or is this unique to me?  It also seems that when I play videos with this addon that it does not add them to the recently watched category.

**Comment by drumdum** [comment]
I've got the exact same problem. After an update to 0.13.10 I could not find my recently viewed category. Therefore I wanted to use add to my list to have my currently viewing series at hand. Any time I want to add anything to my list I get the error in the pastebin:
https://pastebin.com/ckSFi8UQ
I'm on Milhouse libreelec on a RPI3.

**Comment by tiroy1** [comment]
Same error on my OSMC.

**Comment by caphm** [comment]
Adding / removing to my list is working fine on a clean setup. Please try with the newest version and start with a clean slate, i.e. deleting addon data.

**Comment by knappster1** [comment]
@caphm I should be able to try this out in the next couple of days.

**Comment by tiroy1** [comment]
@caphm Unfortunately this still doesn't work. I have uninstalled the Netflix plugin with removing addon settings, installed it again (version 0.13.12 as available in the plugin repository), logged in with my credentials and tried to remove the item from My list. No success.

**Comment by caphm** [comment]
Please provide a debug log with the clean setup. Ideally right after newly installing the add-on. Thanks.

**Comment by knappster1** [comment]
@caphm I am still experiencing the issue as well.  Here is the debug log after clean setup.
https://pastebin.com/gvJ2ZqYQ

**Comment by caphm** [comment]
Please post the full log, as the issue might have its roots before the parts you posted, thanks.

**Comment by knappster1** [comment]
Sure thing.  Full debug log is here:
https://pastebin.com/TLdBAwtS

I believe the log includes loading kodi, installing the addon, then launching, logging in, going to my list, and trying to remove the video from my list.  Let me know if you need additional info.

**Comment by caphm** [comment]
I removed the link to your log, since it contained your Netflix credentials (debug messages by Kodi as to which keys were handled while you were logging in).

The log looks fine so far. The only way I was able to reproduce was by supplying a wrong authURL with the request. But this should be correct, since you can use all other API operations successfully... I'll provide you with a temp release that'll output more debug info. Could take a couple of days though, I'm swamped at work right now.

In the meantime: does it only happen with this particular show? Can you add or remove other items to "my list"?

**Comment by knappster1** [comment]
Whoops.  Is that key just the ESN number?  If you let me know I will cut it out of the log and repost it.

**Comment by caphm** [comment]
No, the authURL is a session specific key that is extracted from the Netflix website data.

Can you please install this modified release and post another debug log: https://github.com/caphm/plugin.video.netflix/archive/debug_383.zip

Thanks.

**Comment by caphm** [comment]
Sorry, there was a typo in a variable name, my bad. I pushed a fix, please download and install the zip again.

**Comment by knappster1** [comment]
I'll give it a try tonight.  Do you have any details on posting debug logs without the authURL info?

**Comment by caphm** [comment]
Posting the authURL ist fine, as it's worthless without the cookie. The reason I removed the link to your log was because one could read your email and password (the credentials you use to log into your Netflix account) out of it.
When debug logging is enabled, Kodi also logs all keypresses. Thus, when you logged into your account, all the keypresses for entering your email and password were also logged.

**Comment by knappster1** [comment]
@caphm, one other thing I noticed recently is that if I make a change to my list from another source, like a web browser, it is not reflected in the kodi plugin until I exit and reopen kodi.  Is there any other info I can share at this time?

**Comment by caphm** [comment]
I suspect it's got something to do with using mutltiple profiles (where you are not using the first one in the list). I'm sorry but this will take some time to fix as I am busy with work currently and can only attend to simple issues. I'll let you know if anything comes up.

**Comment by knappster1** [comment]
Thanks for the response @caphm.  No worries, I just wanted to post my latest observation.  This is relatively minor and it is great to have this plugin so that I dont have to switch inputs, remotes, etc to watch Netflix.  Thanks for your efforts.

**Comment by alexricher** [comment]
I confirm I have the same issue. I do have multiple accounts and I'm not using the first one in the list. Tried clean install with other accounts and got the same error as reported above. Using 0.13.13.

**Comment by chewvodka** [comment]
I'd just like to report that I have this same problem.  I tried reporting and asking for help in the kodi.tv forums here: https://forum.kodi.tv/showthread.php?tid=329767&pid=2787169#pid2787169

I, too, am using a secondary profile when I have this problem.  It works fine with the primary profile.

**Comment by caphm** [comment]
Leave it open for now, as it'll only be fixed once the PR is merged ;) Good to see this is confirmed as fixed, though.

**Comment by sorin-pop** [comment]
@caphm would it be possible to add this fix to the next 0.13 release as well? As I understand, this was fixed only in 0.14.0, but as I see, there are still problems with that heavily refactored version and it is not sure when it will be stable enough to be released, it might still take a lot of time until 0.14.0 is out and stable.

**Comment by budafuko** [comment]
Issue still persists

**Comment by chewvodka** [comment]
'My List' works in the "CastagnaIT" build/fork.  We've been using it at my house for a number of weeks with no issues, multiple profiles. The only function not working is 'Export'.
https://github.com/CastagnaIT/plugin.video.netflix
